### PR TITLE
[Master] Added `Extending User Self Registration and Account Confirmation`

### PR DESCRIPTION
**Pull request by divyaamunugama**
Updates:

- Added the doc as an extension
- Added the APIs included in the doc to our API section

Related issue: https://github.com/wso2/product-is/issues/14967

### CODE REVIEW

File: en/docs/apis/restapis/self-sign-up.yaml [hunk 1]
```diff
@@ -52,34 +52,119 @@ paths:
       summary: |
         Self register user
       description: |
-        This API is used for user self registration.
+        This API is used for user self-registration. <br><br> In addition to the basic self register API, WSO2 provides extensions for self-registration verification. Learn more about the <a href="https://is.docs.wso2.com/en/latest/references/extend/user-mgt/self-register-verification/">extended self registration verification</a> process.
```
Comment (nilminiwso2) [inline en/docs/apis/restapis/self-sign-up.yaml:55]:
```suggestion
        This API is used for user self-registration. <br><br> In addition to the basic self-register API, WSO2 provides extensions for self-registration verification. Learn more about the <a href="https://is.docs.wso2.com/en/latest/references/extend/user-mgt/self-register-verification/">extended self registration verification</a> process.
```

File: en/docs/guides/identity-lifecycles/self-registration-workflow.md [hunk 2]
```diff
@@ -60,13 +60,13 @@ Follow the steps given below to register users for the super tenant, which is `
...
     <td>Lock user account on creation</td>
-    <td>Enable account lock during self registration. The account will be unclocked upon confirmation.</td>
+    <td>Enable account lock during self-registration. The account will be unclocked upon confirmation.</td>
     </tr>
     <tr class="odd">
     <td>Manage notifications sending internally</td>
     <td>
     <p>
-    Select to configure Identity server to send confirmation emails to the user.
+    Select to configure the Identity server to send confirmation emails to the user.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/self-registration-workflow.md:69]:
```suggestion
    Enable the Identity server to send confirmation emails to the user.
```

File: en/docs/guides/identity-lifecycles/self-registration-workflow.md [hunk 3]
```diff
@@ -189,92 +185,86 @@ Now you have set up self registration. Next let's see how you can configure sel
...
 
-3.  Fill in the user details, provide consent to share the requested
-    information and then click **Register**.
-    
-    ![Self sign up form]({{base_path}}/assets/img/guides/self-signup-form.png) 
-        
-    ??? Abstract "Click to see steps on configuring requested attributes for self registration" 
-        
-        -   The attributes that show up on the self sign up page are WSO2 [local dialect]({{base_path}}/guides/dialects/add-claim-mapping/) claims that have the Supported by Default configuration enabled.
+3. Fill in the user details, provide consent to share the requested information and then click **Register**.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-lifecycles/self-registration-workflow.md:245]:
```suggestion
3. Fill in the user details, provide consent to share the requested information, and then click **Register**.
```

File: en/docs/references/extend/user-mgt/self-register-verification.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
+# Verification methods of Self-Registration
+
+WSO2 Identity Server (WSO2 IS) supports self-registration, allowing users to set up their accounts and receive a confirmation when the account is created.
+
+WSO2 Identity Server allows users to confirm their accounts via a user-preferred communication channel. Based on the user preference, the user self-registration process supports email or mobile as the preferred account confirmation channel for the user.
```
Comment (nilminiwso2) [inline en/docs/references/extend/user-mgt/self-register-verification.md:5]: email or mobile? 
in the next line we say SMS and EMAIL?

@divyaamunugama is this right?

Comment (nilminiwso2) [reply en/docs/references/extend/user-mgt/self-register-verification.md:5]: better stick to one term `mobile` or `SMS`

Comment (somindatommy) [reply en/docs/references/extend/user-mgt/self-register-verification.md:5]: It has to be SMS and EMAIL.
When it comes to mobile there can be multiple ways.
- SMS
- Call and ask the call to answer etc

File: en/mkdocs.yml [hunk 5]
```diff
@@ -1445,7 +1446,7 @@ extra:
     - type: linkedin
       link: https://www.linkedin.com/company/wso2
   site_version: 6.0.0
-  base_path: https://is.docs.wso2.com/en/6.0.0
+  base_path: http://127.0.0.1:8000
```
Comment (divyaamunugama) [inline en/mkdocs.yml:1449]:
```suggestion
  base_path: https://is.docs.wso2.com/en/6.0.0
```

File: en/docs/guides/identity-lifecycles/self-registration-workflow.md [hunk 6]
```diff
@@ -324,3 +312,4 @@ curl -X POST -H "Authorization: Basic <Base64Encoded_username:password>" -H "Con
     - [Guide: Invitation Workflow]({{base_path}}/guides/identity-lifecycles/invitation-workflow) 
     - [Guide: Bulk Import Users]({{base_path}}/guides/identity-lifecycles/bulk-import-users)
     - [Guide: Outbound Provisioning]({{base_path}}/guides/identity-lifecycles/outbound-provisioning) 
+    - [References: Self-Registration verification mechanisms]({{base_path}}/references/extend/user-mgt/self-register-verification)
```
Comment (divyaamunugama) [inline en/docs/guides/identity-lifecycles/self-registration-workflow.md:315]:
```suggestion
    - [Guides: Configure Account Confirmation Methods for Self-Registration]({{base_path}}/references/extend/user-mgt/self-register-verification)
```

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**